docs/GettingStarted.md

```diff
@@ -3,47 +3,252 @@
 
 ```bash
 curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
 ```
 
+- Install the [Azure Functions core tools](https://github.com/Azure/azure-functions-core-tools/blob/v4.x/README.md#linux) to work with Azure functions.
 - Check out the GitHub repository in your environment
 
 ``` bash
 git clone https://github.com/microsoft/connected-fleet-refarch.git
 ```
 
-- Open Visual Studio Code from WSL using the argument
+- If using WSL, Open Visual Studio Code from WSL on the code directory
 
 ``` bash
 cd connected-fleet-refarch
 code .
 ```
 
-- Install the following extensions in Visual Studio Code
+- If you want to modify the function code, install the following extensions in Visual Studio Code to make your life easier.
   - [Bicep](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-bicep): Bicep language support for Visual Studio Code
   - [Azure Functions](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-azurefunctions): quickly create, debug, manage, and deploy serverless apps directly from VS Code
 
-## Deployment
+### Deploying the telemetry platform layer
+
+In this step you will:
+
+- Create the test certificates for the Event Grid MQTT broker and test devices.
+- Deploy all required Telemetry Platform resources to Azure using a biceps script.
+- Register 5 test devices to Event Grid as part of the biceps script
+
+#### Telemetry platform deployment artifacts
+
+#### Create test certificates
+
+- Change your directory to ```infra/deployment/TelemetryPlatform```
+
+- Make the scripts executable:
+
+```bash
+chmod 700 generate-root-certificate.sh
+chmod 700 generate-client-certificates.sh
+chmod 700 ./cert-gen/certGen.sh
+```
+
+- Create a test root and intermediate certificate.
+
+> [!Warning]
+> Make sure to run this script **only once** to avoid discrepancies between the generated certificates and the configuration files.
+
+```bash
+./generate-root-certificate.sh
+```
+
+* Create certificates for the test clients
+
+```bash
+./generate-client-certificates.sh
+```
+
+#### Deploy the telemetry platform using biceps
+
+- login to your Azure account and select your subscription.
+
+``` bash
+az login
+```
+- export variables with your preferred resource group names and location.
+
+``` bash
+export RG_TELEMETRYPLATFORM=telemetryplatform
+export RG_FLEETINTEGRATION=fleetintegration
+export LOCATION=eastus
+```
+
+- [Create a resource group](https://learn.microsoft.com/cli/azure/manage-azure-groups-azure-cli#create-a-resource-group) for the deployment in a region
+
+``` bash
+az group create --name ${RG_TELEMETRYPLATFORM} --location ${LOCATION}
+```
+
+- Execute the main.bicep targeting your telemetry platform resource group
+
+``` bash
+az deployment group create --resource-group ${RG_TELEMETRYPLATFORM}--template-file ./main.bicep 
```

Comment:
Optional: when doing Bicep deployments, I find it helpful to say approximately how long it will take to deploy. (When I'm following along I use this as a signal to go to get a coffee or get lunch!)

For me, it took about ~4 minutes for the TelemetryPlatform deployment and ~14 minutes for the FleetIntegration deployment.